src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,453 @@
+synchronous code: if the code within the `try` clause throws an exception,
+the code inside the `catch` clause executes. The only difference is that with
+asynchronous functions you must use the async/await keywords following the 
+syntax you used in the prior section:
+
+[//]: https://gist.github.com/25ade03f0632878a9169209e3cd7bef2
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=25ade03f0632878a9169209e3cd7bef2"
+  frameborder="no"
+  height="525"
+  width="100%" >
+</iframe>
+
+## Practice handling errors
+Use `async` and `await` to accomplish the following:
+
+* Implement a `changeUsername` function that calls the provided asynchronous function
+  `getNewUsername` and returns its result.
+* The `changeUsername` function must catch any errors thrown by `getNewUsername`.
+
+[//]: https://gist.github.com/858f71f0ad0e70051999bcafa41806a3
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=858f71f0ad0e70051999bcafa41806a3&theme=dark"
+  frameborder="no"
+  height="525"
+  width="100%" >
+</iframe>
+
+## Putting it all together
+
+It's time to sum up what you've learned in one final exercise. Write the following:
+
+### Part 1: `addHello` 
+* Write a function `addHello.` that takes a single string
+  argument
+ * `addHello` should return its String argument surrounded by the text ‘Hello <string>!'
+
+### Part 2: `greetUser`
+* Write a function `greetUser` that takes no arguments.
+* To obtain the username, `greetUser` should call the provided async function
+  `getUsername`, which waits for 1 second before returning a string.
+* `greetUser` should create a greeting for the user by calling `addHello,`
+  passing it the username, and returning the result. For example, if the
+  username is ‘Jenny' `greetUser` should create and return the greeting "Hello
+  Jenny!"
+
+### Part 3: `sayGoodbye`
+* Write a function `sayGoodbye` that takes no arguments.
+* Within its body, `sayGoodbye` should call the provided async function
+  `logoutUser` which takes no arguments.
+* `sayGoodbye` should catch any errors thrown by `logoutUser`.
+* If `logoutUser` succeeds, `sayGoodbye` should return the string "<result>
+  Thanks! See you next time!" where <result> is the String value returned by
+  calling `logoutUser`.
+
+[//]: https://gist.github.com/f601d25bc2833c957186e3c6bf71effc
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=f601d25bc2833c957186e3c6bf71effc&theme=dark"
+  frameborder="no"
+  height="525"
+  width="100%">
+</iframe>
+
+## What next?
+
+* Try [other Dart codelabs](/codelabs).
+* Play with [DartPad.]({{site.dartpad}})
+* The [Futures](/tutorials/language/futures) tutorial
+* Read about [Dart's type system](/guides/language/sound-dart) for more examples
+of type signatures with futures. You should
+[prefer signatures in function type annotations](/guides/language/effective-dart/design#prefer-signatures-in-function-type-annotations)
+wherever possible!
```

Comment:
This list is not very parallel. Also, I'd reorder a tad. Perhaps:

* [Asynchronous programming: futures & async-await]() tutorial
* [Dart's type system][] (Includes more examples of type signatures with futures.)
* [Preferring signatures in function type annotations][], from [Effective Dart][]
* [Dart codelabs][]
* [DartPad][]